src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -300,6 +284,56 @@ UpdateRelationsToLocalShardTables(Node *node, List *relationShardList)
 								 relationShardList, QTW_EXAMINE_RTES_BEFORE);
 	}
 
+	if (IsA(node, OnConflictExpr))
+	{
+		OnConflictExpr *onConflict = (OnConflictExpr *) node;
+		Oid constraintId = onConflict->constraint;
+
+		if (constraintId != InvalidOid)
+		{
+			Oid constraintRelationId = InvalidOid;
+
+			HeapTuple tp = SearchSysCache1(CONSTROID, ObjectIdGetDatum(constraintId));
+			if (HeapTupleIsValid(tp))
```

Comment:
again, Is it possible that heaptuple becomes invalid other than buggy cases, should we throw an error for else ?